indexer/services/roundtable/src/tasks/update-affiliate-info.ts

```diff
@@ -0,0 +1,50 @@
+import { logger, stats } from '@dydxprotocol-indexer/base';
+import {
+  PersistentCacheTable, AffiliateInfoTable, PersistentCacheKeys, PersistentCacheFromDatabase,
+} from '@dydxprotocol-indexer/postgres';
+import { DateTime } from 'luxon';
+
+import config from '../config';
+
+const defaultLastUpdateTime: string = '2023-10-26T00:00:00Z';
+
+/**
+ * Update the affiliate info for all affiliate addresses.
+ */
+export default async function runTask(): Promise<void> {
+  try {
+    const start = Date.now();
+    const persistentCacheEntry: PersistentCacheFromDatabase | undefined = await PersistentCacheTable
+      .findById(PersistentCacheKeys.AFFILIATE_INFO_UPDATE_TIME);
+
+    if (!persistentCacheEntry) {
+      logger.info({
+        at: 'update-affiliate-info#runTask',
+        message: `No previous ${PersistentCacheKeys.AFFILIATE_INFO_UPDATE_TIME} found in persistent cache table. Will use default value: ${defaultLastUpdateTime}`,
+      });
+    }
+
+    const lastUpdateTime: DateTime = DateTime.fromISO(persistentCacheEntry
+      ? persistentCacheEntry.value
+      : defaultLastUpdateTime);
+    let windowEndTime = DateTime.utc();
+
+    // During backfilling, we process one day at a time to reduce roundtable runtime.
+    if (windowEndTime > lastUpdateTime.plus({ days: 1 })) {
```

Comment:
this is a pretty expensive operation, I'm concerned that if you processed a day of data it could cause issues, I would recommend starting at 1 hour

Reply:
This was tested during wallet total update on imperator db. For a day it is <1s.